File: array/array-1/1.js
function twoSum(nums, target) {
    const mp = new Map();
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const remaining = target - num;
        if (mp.has(remaining)) {
            return [mp.get(remaining), i];
        }
        mp.set(num, i);
    }
    return [];
}

File: array/array-1/1.py
from typing import List
class Solution:
    def twoSum(self, nums:List[int], target: int) -> List[int]:
        mp = {}
        for i, num in enumerate(nums):
            remaining = target - num
            if remaining in mp:
                return [mp[remaining], i]
            mp[num] = i
        return []